summary/fetcher/data.go

```diff
@@ -9,9 +9,13 @@ import (
 	"github.com/tidepool-org/platform/data"
 )
 
-type DatumCreator func() data.Datum
+type DatumCreator func(typ string) (data.Datum, error)
 type DataCursorFactory func(cursor *mongo.Cursor) DeviceDataCursor
 
+type datumType struct {
```

Comment:
This is fine of course, but might be a good candidate for a anonymous struct, given it's very limited usage.

That or maybe there's a custom unmarshaling interface we can implement for mongo cursors?

But whatever, I'm just thinking out loud, this is fine.